test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerTest.cs

```diff
@@ -59,6 +60,37 @@ public void EnumeratorCacheIsScopedByAssembly()
             Assert.Equal(expectedCallCount, resourceAssembly2.GetManifestResourceStreamCallCount);
         }
 
+        [Fact]
+        public void ResourceLookupBehaviorUseNameIfNotFound()
+        {
+            // Arrange
+            var resourceNamesCache = new ResourceNamesCache();
+            var baseName = "test";
+            var resourceAssembly = new TestAssemblyWrapper("Assembly1");
+            var resourceManager = new TestResourceManager(baseName, resourceAssembly.Assembly);
+            var localizer = new ResourceManagerStringLocalizer(resourceManager, resourceAssembly, baseName, resourceNamesCache, ResourceLookupBehavior.UseNameIfNotFound);
+
+            // Act
+            var value = localizer["test"];
+
+            // Assert
+            Assert.Equal("test", value);
+        }
+
+        [Fact]
+        public void ResourceLookupBehaviorThrowIfNotFound()
```

Comment:
The name should be descriptive of the assertion as well, e.g. `ThrowsForMissingStringWhenResourceLookupBehaviorIsThrowIfNotFound`